Sources/SmithyJSON/Reader/Reader+JSONDeserialization.swift

```diff
@@ -0,0 +1,24 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.Data
+import class Foundation.NSError
+import class Foundation.JSONSerialization
+import class Foundation.NSNull
+
+extension Reader {
```

Comment:
Creates a Reader from JSON data, that can then be used to deserialize the response.

The "catch" clause below emulates current behavior, which deserializes a "default value" of an empty object when the JSON is not valid.